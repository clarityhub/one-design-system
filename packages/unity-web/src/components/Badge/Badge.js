import React from 'react';
import isPropValid from '@emotion/is-prop-valid';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { node, oneOf } from 'prop-types';
import colors from '@clarityhub/unity-core/lib/colors';
import { opacify } from '@clarityhub/unity-core/lib/utilities/color';
import { variants } from '@clarityhub/unity-core/lib/typography';

const Badge = styled('div', {
	shouldForwardProp: prop => isPropValid(prop) && !['type'].includes(prop),
})`
    overflow: visible;
    text-transform: none;
    display: inline-block;

    ${variants.button.string}

    font-weight: 600;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    border: 0 solid transparent;
    padding: 0.1rem 0.4rem;
    font-size: 0.8rem;
    line-height: 1.2;
    border-radius: 2rem;
    position: relative;
    text-transform: uppercase;

    ${({ type }) => {
		switch (type) {
		case 'primary':
			return css`
                    color: ${colors.primary.default};
                    background-color: ${opacify(colors.primary.default, 0.25)};
                `;
		case 'success':
			return css`
                    color: ${colors.success.default};
                    background-color: ${opacify(colors.success.default, 0.25)};
                `;
		case 'danger':
			return css`
                    color: ${colors.danger.default};
                    background-color: ${opacify(colors.danger.default, 0.25)};
                `;
		case 'notification':
			return css`
                    color: ${colors.white.default};
                    background-color: ${colors.notification.default};
                    border-color: ${colors.notification.default};
                `;
		case 'default':
		default:
			return css`
                    color: ${colors.black.default};
                    background-color: ${colors.muted.default};
                    border-color: ${colors.muted.default};
                `;
		}
	}}
`;

/**
 * Hack for docz and react-docgen
 */
const BadgePropTypes = () => <div />;
BadgePropTypes.propTypes = {
	children: node,
	type: oneOf(['primary', 'success', 'danger', 'notification', 'default']),
};
Badge.propTypes = BadgePropTypes.propTypes;
export { BadgePropTypes };

export default Badge;
