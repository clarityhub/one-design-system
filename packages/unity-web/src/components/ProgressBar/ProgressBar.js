import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { number, string, node } from 'prop-types';
import borders from '@clarityhub/unity-core/lib/borders';
import colors from '@clarityhub/unity-core/lib/colors';

const ProgressWrapper = styled.div`
    position: relative;
`;

const ProgressInfo = styled.div`
    display: flex;
    margin-bottom: 0.5rem;
    align-items: center;
    justify-content: space-between;
`;

const ProgressCore = styled.div`
    display: flex;
    overflow: hidden;
    height: 0.8rem;
    margin-bottom: 1rem;
    border-radius: ${borders.borderRadius.rem}rem;
    background-color: ${colors.muted.default};
    box-shadow: inset ${colors.outlineShadow.default};
`;

const ProgressBar = styled.div`
    height: auto;
    border-radius: 0;
    box-shadow: none;
    display: flex;
    flex-direction: column;
    transition: width .6s ease;
    text-align: center;
    white-space: nowrap;
    color: ${colors.white.default};
    background-color: ${colors.primary.default};
    justify-content: center;

    ${({ type }) => {
		switch (type) {
		case 'notification':
			return css`
                    background-color: ${colors.notification.default};
                `;
		case 'danger':
			return css`
                    background-color: ${colors.danger.default};
                `;
		case 'success':
			return css`
                    background-color: ${colors.success.default};
                `;
		case 'primary':
		default:
			return css`
                    background-color: ${colors.primary.default};
                `;
		}
	}}
`;

const Progress = ({ children, progress = 0, type = 'default' }) => (
	<ProgressWrapper>
		<ProgressInfo class="progress-info">
			<div class="progress-label">
				{children}
			</div>
			<div class="progress-percentage">
				<span>{progress}%</span>
			</div>
		</ProgressInfo>
		<ProgressCore>
			<ProgressBar
				type={type}
				role="progressbar"
				aria-valuenow={progress}
				aria-valuemin="0"
				aria-valuemax="100"
				style={{ width: `${progress}%` }}
			/>
		</ProgressCore>
	</ProgressWrapper>
);

Progress.propTypes = {
	children: node,
	progress: number.isRequired,
	type: string,
};

export default Progress;
