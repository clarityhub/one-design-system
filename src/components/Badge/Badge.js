import React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { oneOf } from 'prop-types';

import colors from '../../theme/colors';
import { opacify } from '../../utilities/color';
import { variants } from '../../theme/fonts';

const Badge = styled.div`
    overflow: visible;
    text-transform: none;
    display: inline-block;

    ${variants.button}

    font-weight: 600;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    border: 0 solid transparent;
    padding: 0.1rem 0.4rem;
    font-size: 0.8rem;
    line-height: 1.2;
    border-radius: 2rem;
    position: relative;
    text-transform: uppercase;

    ${({ type }) => {
        switch (type) {
            case 'primary':
                return css`
                    color: ${colors.primary};
                    background-color: ${opacify(colors.primary, 0.3)};
                `;
            case 'success':
                return css`
                    color: ${colors.success};
                    background-color: ${opacify(colors.success, 0.3)};
                `;
            case 'danger':
                return css`
                    color: ${colors.danger};
                    background-color: ${opacify(colors.danger, 0.3)};
                `;
            case 'notification':
                return css`
                    color: ${colors.white};
                    background-color: ${colors.notification};
                    border-color: ${colors.notification};
                `;
            case 'default':
            default:
                return css`
                    color: ${colors.black};
                    background-color: ${colors.muted};
                    border-color: ${colors.muted};
                `;
        }
    }}
`;

/**
 * Hack for docz and react-docgen
 */
const BadgePropTypes = () => <div />;
BadgePropTypes.propTypes = {
    type: oneOf(['primary', 'success', 'danger', 'notification', 'default']),
};
Badge.propTypes = BadgePropTypes;
export { BadgePropTypes };

export default Badge;
