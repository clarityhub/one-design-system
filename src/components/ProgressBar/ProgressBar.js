import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { number, string } from 'prop-types';

import borders from '../../theme/borders';
import colors from '../../theme/colors';

const ProgressWrapper = styled.div`
    position: relative;
    padding-top: 1.5rem;
`;

const ProgressInfo = styled.div`
    display: flex;
    margin-bottom: 0.5rem;
    align-items: center;
    justify-content: space-between;
`;

const ProgressCore = styled.div`
    display: flex;
    overflow: hidden;
    height: 0.8rem;
    margin-bottom: 1rem;
    border-radius: ${borders.borderRadius};
    background-color: #e9ecef;
    box-shadow: inset ${colors.boxShadow};
`;

const ProgressBar = styled.div`
    height: auto;
    border-radius: 0;
    box-shadow: none;
    display: flex;
    flex-direction: column;
    transition: width .6s ease;
    text-align: center;
    white-space: nowrap;
    color: #fff;
    background-color: ${colors.primary};
    justify-content: center;

    ${({ type }) => {
		switch (type) {
		case 'notification':
			return css`
                    background-color: ${colors.notification};
                `;
		case 'danger':
			return css`
                    background-color: ${colors.danger};
                `;
		case 'success':
			return css`
                    background-color: ${colors.success};
                `;
		case 'primary':
		default:
			return css`
                    background-color: ${colors.primary};
                `;
		}
	}}
`;

const Progress = ({ children, progress = 0, type = 'default' }) => (
	<ProgressWrapper>
		<ProgressInfo class="progress-info">
			<div class="progress-label">
				{children}
			</div>
			<div class="progress-percentage">
				<span>{progress}%</span>
			</div>
		</ProgressInfo>
		<ProgressCore>
			<ProgressBar
				type={type}
				role="progressbar"
				aria-valuenow={progress}
				aria-valuemin="0"
				aria-valuemax="100"
				style={{ width: `${progress}%` }}
			/>
		</ProgressCore>
	</ProgressWrapper>
);

Progress.propTypes = {
	progress: number,
	type: string,
};

export default Progress;
