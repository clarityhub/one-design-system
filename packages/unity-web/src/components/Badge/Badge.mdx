---
menu: Components
name: Badges
---

import { Playground, PropsTable } from 'docz';
import Badge, { BadgePropTypes } from './Badge';
import Link from '../Link';

# Badges

## \<Badge /\> component

```js
import Badge from '@clarityhub/unity-web/lib/components/Badge';
```

## Prop Types

<PropsTable of={BadgePropTypes} />

## Examples

### Sizes

<Playground>
    <Badge>1</Badge>
    <br />
    <br />
    <Badge>new</Badge>
    <br />
    <br />
    <Badge>1 new</Badge>
</Playground>

### Colors
<Playground>
    <Badge type="default">10</Badge>
    <br />
    <br />
    <Badge type="primary">10</Badge>
    <br />
    <br />
    <Badge type="success">10</Badge>
    <br />
    <br />
    <Badge type="danger">10</Badge>
    <br />
    <br />
    <Badge type="notification">10</Badge>
    <br />
    <br />
</Playground>

### As Links
<Playground>
    <Link type="text" href="#"><Badge type="default">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="primary">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="success">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="danger">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="notification">10</Badge></Link>
    <br />
    <br />
</Playground>
