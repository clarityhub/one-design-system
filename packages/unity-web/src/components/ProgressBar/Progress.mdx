---
menu: Components
name: ProgressBar
---

import { Component } from 'react';
import { Playground, PropsTable } from 'docz';
import ProgressBar from './ProgressBar';
import Badge from '../Badge';

# ProgressBar

Used to display the linear progress of an action. Any children components will be rendered above the Progress Bar

```js
import ProgressBar from '@clarityhub/unity-web/lib/components/ProgressBar';
```

## Prop Types

<PropsTable of={ProgressBar} />

## Examples

### Colors and Percentages

<Playground>
    <ProgressBar primary progress={10} type="default">
        <Badge type="primary">Uploading...</Badge>
    </ProgressBar>
    <br />
    <br />
    <ProgressBar progress={30} type="primary" />
    <br />
    <br />
    <ProgressBar primary progress={100} type="success">
        <Badge type="success">Task Completed</Badge>
    </ProgressBar>
    <br />
    <br />
    <ProgressBar primary progress={97} type="danger">
        <Badge type="danger">Upload Failed</Badge>
    </ProgressBar>
    <br />
    <br />
    <ProgressBar progress={100} type="notification" />
    <br />
    <br />
</Playground>


<Playground>
    {() => {
        class AnimatedProgress extends Component {
            constructor(props) {
                super(props);
                this.state = {
                    progress: parseInt(+new Date() / 100 % 100),
                };
            }

            componentDidMount() {
                this.timer = setInterval(() => {
                    this.setState({
                        progress: parseInt(+new Date() / 100 % 100),
                    })
                }, 1000);
            }

            componentWillUnmount() {
                if (this.timer) {
                    clearInterval(this.timer);
                }
            }

            render() {
                const { progress } = this.state;

                return (
                    <ProgressBar primary progress={progress} type="default">
                        <Badge type="primary">Uploading...</Badge>
                    </ProgressBar>
                );
            }
        }

        return <AnimatedProgress />;
    }}
</Playground>