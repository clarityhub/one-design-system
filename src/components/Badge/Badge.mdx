---
menu: Components
name: Badges
---

import { Playground, PropsTable } from 'docz';
import Badge, { BadgePropTypes } from './Badge';

# Badges

## \<Badge /\> component

```js
import Badge from '@clarityhub/one/components/Badge';
```

## Prop Types

<PropsTable of={BadgePropTypes} />

## Examples

### Sizes

<Playground>
    <Badge>1</Badge>
    <br />
    <br />
    <Badge>new</Badge>
    <br />
    <br />
    <Badge>1 new</Badge>
</Playground>

### Colors
<Playground>
    <Badge type="default">10</Badge>
    <br />
    <br />
    <Badge type="primary">10</Badge>
    <br />
    <br />
    <Badge type="success">10</Badge>
    <br />
    <br />
    <Badge type="danger">10</Badge>
    <br />
    <br />
    <Badge type="notification">10</Badge>
    <br />
    <br />
</Playground>
