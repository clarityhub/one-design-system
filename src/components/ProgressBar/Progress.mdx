---
menu: Components
name: ProgressBar
---

import { Playground, PropsTable } from 'docz';
import ProgressBar from './ProgressBar';

# ProgressBar

## \<ProgressBar /\> component

```js
import ProgressBar from '@clarityhub/one/components/ProgressBar';
```

## Prop Types

<PropsTable of={ProgressBar} />

## Examples

### Colors and Percentages

<Playground>
    <ProgressBar progress={10} type="default" />
    <br />
    <br />
    <ProgressBar progress={30} type="primary" />
    <br />
    <br />
    <ProgressBar progress={60} type="success" />
    <br />
    <br />
    <ProgressBar progress={80} type="danger" />
    <br />
    <br />
    <ProgressBar progress={100} type="notification" />
    <br />
    <br />
</Playground>
