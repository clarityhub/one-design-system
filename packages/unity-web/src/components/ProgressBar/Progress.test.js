import React from 'react';
import renderer from 'react-test-renderer';

import Badge from '../Badge';
import ProgressBar from './ProgressBar';

describe('<ProgressBar />', () => {
	it('renders progress bar', () => {
		const node = (
			<ProgressBar primary progress={10} type="default">
				<Badge type="primary">Uploading...</Badge>
			</ProgressBar>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});