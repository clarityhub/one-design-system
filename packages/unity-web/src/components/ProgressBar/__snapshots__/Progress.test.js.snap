// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders progress bar 1`] = `
<div
  className="css-13e9g95-ProgressWrapper e1u0vuor0"
>
  <div
    className="css-ureerz-ProgressInfo e1u0vuor1"
  >
    <div>
      <div
        className="css-1bqee8j-Badge-Badge e3hul5p0"
      >
        Uploading...
      </div>
    </div>
    <div>
      <span>
        10
        %
      </span>
    </div>
  </div>
  <div
    className="css-qmk895-ProgressCore e1u0vuor2"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={10}
      className="css-o9hhfa-ProgressBar-ProgressBar e1u0vuor3"
      role="progressbar"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;
